src/jira/jira_data_mappings.bal

```diff
@@ -19,78 +19,77 @@
-
     map<json> target = {};
-
-    target["key"] = request?.key != EMPTY_STRING ? request?.key : null;
-    target["name"] = request?.name != EMPTY_STRING ? request?.name : null;
-    target["projectTypeKey"] = request?.projectTypeKey != EMPTY_STRING ? request?.projectTypeKey : null;
-    target["projectTemplateKey"] = request?.projectTemplateKey != EMPTY_STRING ? request?.projectTemplateKey : null;
-    target["description"] = request?.description != EMPTY_STRING ? request?.description : null;
-    target["lead"] = request?.lead != EMPTY_STRING ? request?.lead : null;
-    target["url"] = request?.url != EMPTY_STRING ? request?.url : null;
-    target["assigneeType"] = request?.assigneeType != EMPTY_STRING ? request?.assigneeType : null;
-    target["avatarId"] = request?.avatarId != EMPTY_STRING ? request?.avatarId : null;
-    target["issueSecurityScheme"] = request?.issueSecurityScheme != EMPTY_STRING ? request?.issueSecurityScheme : null;
-    target["permissionScheme"] = request?.permissionScheme != EMPTY_STRING ? request?.permissionScheme : null;
-    target["notificationScheme"] = request?.notificationScheme != EMPTY_STRING ? request?.notificationScheme : null;
-    target["categoryId"] = request?.categoryId != EMPTY_STRING ? request?.categoryId : null;
-
+    target["key"] = validateStringValue(request?.key);
+    target["name"] = validateStringValue(request?.name);
+    target["projectTypeKey"] = validateStringValue(request?.projectTypeKey);
+    target["projectTemplateKey"] = validateStringValue(request?.projectTemplateKey);
+    target["description"] = validateStringValue(request?.description);
+    target["lead"] = validateStringValue(request?.lead);
+    target["url"] = validateStringValue(request?.url);
+    target["assigneeType"] = validateStringValue(request?.assigneeType);
+    target["avatarId"] = validateStringValue(request?.avatarId);
+    target["issueSecurityScheme"] = validateStringValue(request?.issueSecurityScheme);
+    target["permissionScheme"] = validateStringValue(request?.permissionScheme);
+    target["notificationScheme"] = validateStringValue(request?.notificationScheme);
+    target["categoryId"] = validateStringValue(request?.categoryId);
     return target;
 }
 
 function jsonToProjectSummary(json request) returns ProjectSummary {
-
     ProjectSummary target = {};
-
     target.resourcePath = request.self.toString() ;
     target.id = request.id.toString() ;
     target.key = request.key.toString() ;
     target.name = request.name.toString() ;
-    target.description = request.description != null ? request.description.toString()  : EMPTY_STRING;
+    target.description = validateJsonValue(request.description);
     target.projectTypeKey = request.projectTypeKey.toString() ;
-    target.category = request.projectCategory != null ? request.projectCategory.name.toString()  : EMPTY_STRING;
+    if (request.projectCategory != ()) {
+        target.category = request.projectCategory.name.toString();
```

Comment:
Can you be sure that  `request.projectCategory.name` is not null? because you only check the request.projectCategory
```suggestion
        target.category = validateJsonValue(request.projectCategory.name);
```